js/ui/appIconBar.js

```diff
@@ -1004,6 +1036,23 @@ const AppIconBar = new Lang.Class({
         Main.overview.connect('showing', Lang.bind(this, this._updateActiveApp));
         Main.overview.connect('hidden', Lang.bind(this, this._updateActiveApp));
 
+
+        for (let index = 0; index < 8; index++) {
+            let fullName = 'activate-icon-' + (index + 1);
+            Main.wm.addKeybinding(fullName,
+                                  new Gio.Settings({ schema: SHELL_KEYBINDINGS_SCHEMA }),
+                                  Meta.KeyBindingFlags.NONE,
+                                  Shell.KeyBindingMode.NORMAL |
+                                  Shell.KeyBindingMode.OVERVIEW,
+                                  this._activateNthApp.bind(this, index));
```

Comment:
Ohhh got it, that's the prefix postfix problem. I'd rather not include keybinding handler args in `activateNthApp`.

Didn't want to inline cause my linter was complaining about inlining in a for loop, and there are tons of pitfalls there http://stackoverflow.com/questions/21510331/functions-within-loops-why-are-they-considered-an-error.

I'd say either leave as is, or something like
```js
[1, 2, 3, 4, 5, 6, 7, 8].forEach(function(index) {
    let fullName = 'activate-icon-' + (index + 1);
    let callback = Lang.bind(this, function () {
        this._activateNthApp(index);
    });
    Main.wm.addKeybinding(fullName,
                          new Gio.Settings({ schema: SHELL_KEYBINDINGS_SCHEMA }),
                          Meta.KeyBindingFlags.NONE,
                          Shell.KeyBindingMode.NORMAL |
                          Shell.KeyBindingMode.OVERVIEW,
                          callback);
});
```

What you think?